### PR TITLE
Improve info around encoding

### PR DESCRIPTION
**Pull request by kenchris**
@zolkis @beaufortfrancois

***
<a href="https://pr-preview.s3.amazonaws.com/kenchris/web-nfc/pull/376.html" title="Last updated on Oct 16, 2019, 1:43 PM UTC (1a86a08)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-nfc/376/d1416bd...kenchris:1a86a08.html" title="Last updated on Oct 16, 2019, 1:43 PM UTC (1a86a08)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -3979,24 +3979,45 @@ <h3>Receiving and parsing content</h3>
...
-  </p>
+  <aside class="note" data-link-for="NDEFRecord">
+    <p>
+      The Unicode standard defines multiple encodings like UTF-8, UTF-16 and UTF-32.
+      UTF-8 is the preferred encoding on the web, and it has the advantage
+      that it is endianness agnostic, as code points are represented in single bytes.
+    </p>
+    <p>
+      The NDEF text records allow the text to be encoded as either UTF-8 and UTF-16.
+      Generally, it is preferred to use UTF-8, but integration with existing systems
```
Comment (beaufortfrancois) [inline index.html]: Core NFC uses UTF-16 when creating a text record so we may want to skip this part.

File: index.html [hunk 2]
```diff
@@ -3979,24 +3979,45 @@ <h3>Receiving and parsing content</h3>
...
+      may requires UTF-16.
+    </p>
+    <p>
+      The byte order of NDEF is big endian (BE), which means that everything is read
```
Comment (beaufortfrancois) [inline index.html]: Can we source this?

File: index.html [hunk 3]
```diff
@@ -3979,24 +3979,45 @@ <h3>Receiving and parsing content</h3>
...
+      back with big endian byte ordering.
+    </p>
+    <p>
+      For UTF-16, byte order matters as it might differ between reading and writing
+      due to each code point spanning two bytes. For this reason UTF-16 encoded text
+      usually contains a byte order mark (BOM) which is written as `0xFEFF`. This means
```
Comment (beaufortfrancois) [inline index.html]: Nit: "Byte Order Mark"

**Review by beaufortfrancois — APPROVED** [review]
LGTM with nits

File: index.html (hunk 1 above)
Comment (kenchris) [reply index.html]: On the web UTF-8 is definitely preferred. How do you want to rephrase?

File: index.html [hunk 4]
```diff
@@ -3979,24 +3979,45 @@ <h3>Receiving and parsing content</h3>
...
+  <aside class="note" data-link-for="NDEFRecord">
+    <p>
+      The Unicode standard defines multiple encodings like UTF-8, UTF-16 and UTF-32.
+      UTF-8 is the preferred encoding on the web, and it has the advantage
+      that it is endianness agnostic, as code points are represented in single bytes.
+    </p>
+    <p>
+      The NDEF text records allow the text to be encoded as either UTF-8 and UTF-16.
+      Generally, it is preferred to use UTF-8, but integration with existing systems
+      may requires UTF-16.
```
Comment (beaufortfrancois) [inline index.html]: Nit: "require"

File: index.html [hunk 5]
```diff
@@ -3979,24 +3979,45 @@ <h3>Receiving and parsing content</h3>
...
-    the encoding.
-  </p>
+  <aside class="note" data-link-for="NDEFRecord">
+    <p>
+      The Unicode standard defines multiple encodings like UTF-8, UTF-16 and UTF-32.
+      UTF-8 is the preferred encoding on the web, and it has the advantage
+      that it is endianness agnostic, as code points are represented in single bytes.
+    </p>
+    <p>
+      The NDEF text records allow the text to be encoded as either UTF-8 and UTF-16.
```
Comment (beaufortfrancois) [inline index.html]: UTF-8 and UTF-16 => UTF-8 or UTF-16.

File: index.html (hunk 2 above)
Comment (kenchris) [reply index.html]: Its in the NDEF specs

File: index.html (hunk 1 above)
Comment (beaufortfrancois) [reply index.html]: Maybe something like "UTF-8 is the preferred format on the Web. This is why text records in Web NFC defaults to 
UTF-8"

File: index.html (hunk 3 above)
Comment (kenchris) [reply index.html]: encoding writes it lower case ... let me do it differently

**Comment by beaufortfrancois** [comment]
Still LGTM.
Let's wait for @zolkis to approve as well.

**Comment by zolkis** [comment]
LGTM (will struggle with merging into my PR, but anyway).

**Pull request merged**